docs/services-web-app.md

```diff
@@ -12,3 +29,255 @@ access to private repos and increases API quotas. To set up a personal access to
+Statically Generated Pages" section in
+[NextJs Authentication Docs](https://nextjs.org/docs/authentication) You may make use of the
+`useAuth()` hook and `RequireAuth` component in your implementations:
+
+```
+  import { useAuth } from 'contexts/auth'
+  import { useEffect } from 'react'
+
+  import RequireAuth from '@/components/auth/require-auth'
+
+  import FourOhFour from '../404'
+
+  const ProtectedStaticPage = () => {
+    const { isAuthenticated, loading, user } = useAuth()
+
+    useEffect(() => {
+      if (!loading && isAuthenticated) {
+        // fetch protected data here
+      }
+    }, [loading, isAuthenticated])
+
+    return loading ? null : (
+      // show page if user is authenticated and email ends with ibm.com, else show 404 page
+      <RequireAuth
+        fallback={FourOhFour}
+        isAuthorized={isAuthenticated && user?.email?.endsWith('ibm.com')}
+      >
+        <>
+          // Your Page Content Here
+        </>
+      </RequireAuth>
+    )
+  }
+
+  // do NOT fetch protected data here
+  export async function getStaticProps(/* context */) {
+    return {
+      props: {} // will be passed to the page component as props
+    }
+  }
+
+  export default ProtectedStaticPage
+```
+
+Note: DO NOT SERVE/FETCH PROTECTED INFO THROUGH `getStaticProps`, as these will be served
+automatically when the route is accessed regardless of authentication.
+
+For a working example of a protected static page, (run the app securely)[#running-app-securely] and
+visit [https://localhost/samples/protectedStaticPage](https://localhost/samples/protectedStaticPage)
+
+## Retrieving User's Data
+
+### Server Side Rendered Pages
+
+#### retrieveUser
+
+The `retrieveUser` function will return the current user instance or null if user is not logged in;
+note this function is asynchronous. This function can only be called server-side (i.e., in the
+context of getServerSideProps):
+
+```
+import { retrieveUser } from '@/utils/retrieveUser'
+...
+export const getServerSideProps = async (/* context */) => {
+  const user = await retrieveUser(context)
+  // Your normal `getServerSideProps` code here
+  return {
+    props: {
+      user
+      // Your custom props here
+    }
+  }
+}
+```
+
+_Note_: if you are wrapping your `getServerSideProps` with `getPropsWithAuth` and your
+`authorizationChecker` function calls `retrieveUser()`, the obtained user (if any) will be injected
+into the pages's props; you can access it on `props.user`
+
+### UseAuth() Hook
+
+From a React Component, you can access the user's information via the `user` property in the
+`useAuth` hook. This hook also exposes other info such as:
+
+- user: object containing user's data
+- loading: boolean indicating whether user is still being retrieved or not
+- isAuthenticated: booleand indicating whether the user has correctly authenticated or not
+- login: function to login user
+- logout: function to logout user
+
+```
+ import { useAuth } from 'contexts/auth'
```

Comment:
See comment on the feat(auth) PR about this. Might need a tweak depending on the result of that.